### ISSUE TITLE
Integrate MOTOR-T Foundation Model with EHRSHOT Multi-Event Survival Data

### ISSUE THREAD

**Issue by dahlem** [open-disease-risk/sat#40]
# Integrate MOTOR-T Foundation Model with EHRSHOT Multi-Event Survival Data

## Objective
Integrate Stanford's MOTOR-T medical foundation model into the SAT framework and prepare EHRSHOT benchmark data as a multi-event survival cohort for training survival models on discharge-anchored clinical events.

## Background
- **MOTOR-T**: Stanford's foundation model pre-trained on medical code sequences, optimized for EHR data
- **EHRSHOT**: Benchmark dataset with rich clinical event sequences suitable for multi-event survival analysis
- **Goal**: Leverage pre-trained medical representations for improved survival prediction on multiple clinical outcomes

### 1. MOTOR-T Model Integration
- [ ] Add MOTOR-T transformer configuration (`conf/transformers/motor_t.yaml`)
  - Configure model loading from HuggingFace hub (`StanfordShahLab/motor-t-base`)
  - Set up trust_remote_code flag for custom model code
  - Define architecture parameters (768 hidden dims, 12 layers, 496 context window)
- [ ] Create MOTOR-T tokenizer configuration (`conf/tokenizers/ehrshot_motor.yaml`)
  - Configure medical code tokenization with whitespace splitting
  - Set max sequence length to 496 (MOTOR-T's context window)
  - Define special tokens for medical sequences

### 2. EHRSHOT Data Preparation Pipeline
- [ ] Implement EHRSHOT cohort parser (`sat/data/dataset/ehrshot_cohort.py`)
  - Parse patient sequences with 7 discharge-anchored events
  - Structure as multi-event survival data
  - Handle censoring and event indicators
- [ ] Create data loading configuration (`conf/data/load/ehrshot_motor.yaml`)
  - Define multi-event columns (durations, events)
  - Set up train/valid/test splits
  - Configure JSON lines data format
- [ ] Implement data parsing logic (`sat/data/dataset/parse_ehrshot_motor.py`)
  - Transform raw EHRSHOT data to multi-event format
  - Generate 7 simultaneous survival outcomes per patient

### 3. Experiment Configuration
- [ ] Create base experiment defaults (`conf/experiments/ehrshot_motor/defaults.yaml`)
  - Set dataset parameters (14,251 patients, 7 events)
  - Configure MOTOR-T model dimensions
  - Define training hyperparameters
- [ ] Add survival task configuration (`conf/experiments/ehrshot_motor/survival.yaml`)
  - Configure multi-event survival heads
  - Set loss weights and evaluation metrics
  - Define survival-specific architecture parameters
- [ ] Create multi-task learning config (`conf/experiments/ehrshot_motor/mtl.yaml`)
  - Set up task head configurations
  - Define shared layer parameters

### 4. Framework Enhancements
- [ ] Update label transformation for multi-event data
  - Handle nested array structures for multiple events
  - Generate appropriate time discretization cuts
  - Support datasets with high censoring rates
- [ ] Ensure trainer configuration compatibility
  - Create default trainer config if missing
  - Set appropriate batch sizes for medical sequences
- [ ] Validate data pipeline end-to-end
  - Verify tokenizer training on medical codes
  - Check label transformation output format
  - Ensure compatibility with existing multi-event models

## Expected Outcomes
- Functional pipeline for MOTOR-T + EHRSHOT integration
- Support for 7 simultaneous survival outcomes
- Proper handling of medical code sequences
- Ready-to-train configuration for survival analysis

## Acceptance Criteria
- [ ] Data preparation completes without errors
- [ ] Tokenizer successfully processes medical code sequences  
- [ ] Label transformation generates correct multi-event format (28 columns)
- [ ] Model initialization works with pre-trained MOTOR-T weights
- [ ] Training pipeline runs end-to-end

## Technical Considerations
- Multi-event data structure requires careful array handling
- High censoring rates may require special quantile computation
- Medical codes need appropriate tokenization strategy
- Pre-trained model integration requires trust_remote_code flag

## Testing Commands
```bash
# Verify each pipeline step
poetry run python -m sat.prepare_data experiments=ehrshot_motor/survival
poetry run python -m sat.train_tokenizer experiments=ehrshot_motor/survival
poetry run python -m sat.train_labeltransform experiments=ehrshot_motor/survival
poetry run python -m sat.finetune experiments=ehrshot_motor/survival
```

## References
- [MOTOR-T Model](https://huggingface.co/StanfordShahLab/motor-t-base)
- [EHRSHOT Benchmark](https://www.physionet.org/content/ehrshot/1.0/)
- SAT Framework Documentation
